### PR TITLE
Notes & enable pandas options

### PR DESCRIPTION
**Pull request by holdenk**
Add some notes and enable the pandas options required for display. Also only sample if auto_sample isn't set to false.

### CODE REVIEW

File: sparkviz/__init__.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
 import pyspark.sql as sd
 from pyspark.storagelevel import StorageLevel
 import pandas as pd
+# Turn on pandas application/vnd.dataresource+json for use with
+# https://github.com/nteract/nteract/issues/1526
+pd.options.display.html.table_schema = True
```
Comment (rgbkrk) [inline sparkviz/__init__.py:23]: There is a part of me that feels bad sneaking in side effects that push the direction I want. 😄

File: sparkviz/__init__.py [hunk 2]
```diff
@@ -32,22 +35,27 @@ def special_show(self, n=2000, truncate=False, vertical=False, auto_sample=True,
...
   """A special fake class for DataFrame results. Wraps a pandas df & a spakr df."""
 
-  def __init__(self, pdf, sdf):
+  def __init__(self, pdf, sdf, sampled):
     self.pdf = pdf
     self.sdf = sdf
+    self.sampled = sampled
     from IPython.display import display
+    # TODO(communicate if we've sampled data to the front end?)
     self.display = display(pdf, display_id=True)
```
Comment (rgbkrk) [inline sparkviz/__init__.py:57]: It just occurred to me that we could indicate sampled data in another way, which is `metadata` on the displayed result.

```python
display(
    JSON({'a': 'b'}),
    metadata={
        "application/json": {
            "expanded": True
        }
    }
)
```

screen shot 2017-12-22 at 8 13 17 am

File: sparkviz/__init__.py (hunk 1 above)
Comment (holdenk) [reply sparkviz/__init__.py:23]: Legit, I'll revert it out when we do this with the metadata instead of just using pandas.

Comment (rgbkrk) [reply sparkviz/__init__.py:23]: I though there was a way to coerce pandas to give you this representation, but it appears that `df._repr_data_resource_()` is a noop if `table_schema` isn't set.

**Pull request merged**